enterprise_access/apps/customer_billing/stripe_event_handlers.py

```diff
@@ -115,6 +118,111 @@ def link_event_data_to_checkout_intent(event, checkout_intent):
     event_data.save()
 
 
+def handle_pending_update(subscription_id: str, checkout_intent_id: int, pending_update):
+    """
+    Log pending update information for visibility.
+    Assumes a pending_update is present.
+    """
+    # TODO: take necessary action on the actual SubscriptionPlan and update the CheckoutIntent.
+    logger.warning(
+        "Subscription %s has pending update: %s. checkout_intent_id: %s",
+        subscription_id,
+        pending_update,
+        checkout_intent_id,
+    )
+
+
+def handle_trial_cancellation(checkout_intent: CheckoutIntent, checkout_intent_id: int, subscription_id: str, trial_end):
+    """
+    Send cancellation email for a trial subscription that has just transitioned to canceled.
+    Assumes caller validated status transition and presence of trial_end.
+    """
+    logger.info(
+        f"Subscription {subscription_id} transitioned to 'canceled'. "
+        f"Queuing trial cancellation email for checkout_intent_id={checkout_intent_id}"
+    )
+
+    send_trial_cancellation_email_task.delay(
+        checkout_intent_id=checkout_intent.id,
+        trial_end_timestamp=trial_end,
+    )
+
+
+def future_plans_of_current(current_plan_uuid: str, plans: list[dict]) -> list[dict]:
+    """
+    Return plans that are future renewals of the current plan, based on prior_renewals linkage.
+    """
+    def is_future_of_current(plan_dict):
+        if str(plan_dict.get('uuid')) == current_plan_uuid:
+            return False
+        for renewal in plan_dict.get('prior_renewals', []) or []:
+            if str(renewal.get('prior_subscription_plan_id')) == current_plan_uuid:
+                return True
+        return False
+
+    return [p for p in plans if is_future_of_current(p)]
+
+
+def cancel_all_future_plans(enterprise_uuid: str, reason: str = 'delayed_payment', subscription_id_for_logs: str | None = None) -> list[str]:
+    """
+    Deactivate (cancel) all future plans for the current plan of the given enterprise.
+
+    Returns list of deactivated plan UUIDs. Logs warnings/info for observability.
+    """
```

Comment:
This approach exposes us to liability because it is sensitive to any future change we might make to allow SubscriptionPlans to exist aside from the stripe-managed ones, or any change to allow multiple stripe-managed SubscriptionPlans.  I'd recommend taking into account some other data to join to make sure the plans being deactivated are somehow related to the stripe Subscription.